src/main/java/jewellery/inventory/service/ResourceInUserService.java

```diff
@@ -128,12 +129,23 @@ private ResourcesInUserResponseDto getResourceInUserResponse(UUID userId, UUID r
     return null;
   }
 
-  private ResourceInUser addResourceToUser(User user, Resource resource, Double quantity) {
-    ResourceInUser resourceInUser =
-        findResourceInUser(user, resource.getId())
-            .orElseGet(() -> createAndAddNewResourceInUser(user, resource, 0));
+  private void addResourceToUserToTransfer(User user, Resource resource, Double quantity) {
+    ResourceInUser resourceInUser = getResourceInUser(user, resource);
+    resourceInUser.setQuantity(resourceInUser.getQuantity() + quantity);
+    resourceInUserRepository.save(resourceInUser);
```

Comment:
This method isn't very different than the one that just adds resource to user when doing top-up (a.k.a. delivery). We should consider reusing the existing logic and just extend it for the top-up logic.